File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <map>
#include <vector>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> m;
    
    // for(string s : participant) {
    //     m[s] = 0;
    // }
    
    for(string s : participant) {
        m[s]++;
    }
    
    for(string s : completion) {
        m[s]--;
    }
    
    string answer = "";
    for(string s : participant) {
        if(m[s] != 0) {
            answer = s;
        }
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    
    const map = {};
    for(const name of participant) {
        map[name] = 0;
    }
    for(const name of participant) {
        map[name]++;
    }
    for(const name of completion) {
        map[name]--;
    }
    
    for(const key in map) {
        if(map[key] != 0) {
            answer = key;
        }
    }
    return answer;
}